staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -930,6 +930,15 @@ func (e *Store) Delete(ctx context.Context, name string, deleteValidation rest.V
 	// TODO: remove the check, because we support no-op updates now.
 	if graceful || pendingFinalizers || shouldUpdateFinalizers {
 		err, ignoreNotFound, deleteImmediately, out, lastExisting = e.updateForGracefulDeletionAndFinalizers(ctx, name, key, options, preconditions, deleteValidation, obj)
+		// Update the preconditions.ResourceVersion if set since we updated the object.
+		if err == nil && deleteImmediately && preconditions.ResourceVersion != nil {
+			accessor, err = meta.Accessor(out)
+			if err != nil {
+				return out, false, kubeerr.NewInternalError(err)
+			}
+			resourceVersion := accessor.GetResourceVersion()
+			preconditions.ResourceVersion = &resourceVersion
```

Comment:
if `preconditions.ResourceVersion` is set, that also means `options.Preconditions.ResourceVersion` is set, right? do we need to update that as well? if so, do we need to copy delete options first to avoid mutating what was passed in?

Reply:
We don't need to copy delete options, we don't pass it anywhere, it dead-ends in this function and we access the variables directly as we need them (we set preconditions earlier in the function and never refer to the ones in options again).

With the test I've added we should be safe against regression.  Delete options is specifically for this function, so not sure what the benefit of copying would be at this time.

Reply:
got it, I misread the later use of `options.DryRun` and thought we were passing the entire options struct into subcalls